### PR TITLE
Add flutter clientId to GA dimension8.

### PR DESCRIPTION
**Pull request by terrylucas**
- Ensure .flutter never created by DevTools.

### CODE REVIEW

File: packages/devtools_app/lib/src/ui/analytics.dart [hunk 1]
```diff
@@ -215,17 +225,22 @@ Future<bool> get isFlutterGAEnabled async {
...
-      // Return value 'null' implies Flutter tool has never been run return null.
-      clientId = json.decode(resp.responseText);
-    } else {
-      _logWarning(resp, server.apiGetFlutterGAClientId);
+    // Test if Flutter is enabled (or if Flutter Tool ever run if not enabled is
+    // false, we don't want to be the first to create a ~/.flutter file.
+    if (await isFlutterGAEnabled) {
+      final resp = await _request(server.apiGetFlutterGAClientId);
+      if (resp?.status == HttpStatus.ok) {
+        // Return value 'null' implies Flutter tool has never been run return null.
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/ui/analytics.dart]: comment outdated: r/return null/return empty string.

File: packages/devtools_app/lib/src/ui/analytics.dart [hunk 2]
```diff
@@ -215,17 +225,22 @@ Future<bool> get isFlutterGAEnabled async {
...
 
   if (isDevToolsServerAvailable) {
-    final resp = await _request(server.apiGetFlutterGAClientId);
-    if (resp?.status == HttpStatus.ok) {
-      // Return value 'null' implies Flutter tool has never been run return null.
-      clientId = json.decode(resp.responseText);
-    } else {
-      _logWarning(resp, server.apiGetFlutterGAClientId);
+    // Test if Flutter is enabled (or if Flutter Tool ever run if not enabled is
+    // false, we don't want to be the first to create a ~/.flutter file.
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/ui/analytics.dart]: nit: end parenthesis in comment

Comment (terrylucas) [reply packages/devtools_app/lib/src/ui/analytics.dart]: Was changing comment  - now its done. Good catch.

File: packages/devtools_app/lib/src/ui/analytics.dart (hunk 1 above)
Comment (terrylucas) [reply packages/devtools_app/lib/src/ui/analytics.dart]: You started looking before I requested anyone :-), still correcting.
Done.

File: packages/devtools_app/lib/src/ui/analytics.dart (hunk 2 above)
Comment (terrylucas) [reply packages/devtools_app/lib/src/ui/analytics.dart]: Good catch I fixed another comment not this one. Thanks.

**Pull request merged**